public/app/plugins/datasource/loki/language_provider.ts

```diff
@@ -331,11 +333,12 @@ export default class LokiLanguageProvider extends LanguageProvider {
     return { context, suggestions };
   }
 
-  async importQueries(queries: LokiQuery[], datasourceType: string): Promise<LokiQuery[]> {
+  async importQueries(queries: DataQuery[], originDataSource: DataSourceApi): Promise<LokiQuery[]> {
+    const datasourceType = originDataSource.meta.id;
     if (datasourceType === 'prometheus') {
       return Promise.all(
         queries.map(async (query) => {
-          const expr = await this.importPrometheusQuery(query.expr);
+          const expr = await this.importPrometheusQuery((query as PromQuery).expr);
```

Comment:
Not related to this PR, but we could (after this is merged) also move this to `./importing` similarly as you do with Graphite. I also like how much cleaner is `fromGraphite`. I would like to see the same for Prometheus.

Reply:
Yep. It's quite inconsistent now between data sources. Maybe I'll create an issue to discuss it?  (#33458) The importing is still tightly coupled though and requires importing files from other data sources but at least the logic is extracted. Maybe once we have one more `.importQueries` implementation we could think of some more generic component to translate between different query languages. I didn't want to overcomplicate it at this stage.